utils/src/utils/apiUtils.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Extension, extensions } from "vscode";
+import { localize } from "vscode-nls";
+import type { AzureExtensionApi, AzureExtensionApiProvider } from "../../api";
+import { AzureHostExtensionApi } from "../../hostapi";
+
+export async function getResourceGroupsApi(apiVersionRange: AzureExtensionApi['apiVersion']): Promise<AzureHostExtensionApi>;
+export async function getResourceGroupsApi<T extends AzureExtensionApi>(apiVersionRange: string): Promise<T> {
```

Comment:
The RG `getApi()` API has an options argument where extensions can pass in their ID for telemetry purposes (which we should encourage).  We should mirror that here.

Reply:
Planning on doing that in a follow up PR. I probably shouldn't have included these api utils in this PR to begin with.